pkg/api/v1alpha2/api.proto

```diff
@@ -0,0 +1,461 @@
+     */
+    rpc GetExperimentList(GetExperimentListRequest) returns (GetExperimentListReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetExperimentList"
+        };
+    };
+
+    /** 
+     * Update Status of a experiment.
+     */
+    rpc UpdateExperimentStatus(UpdateExperimentStatusRequest) returns (UpdateExperimentStatusReply){
+        option (google.api.http) = {
+            put: "/api/Manager/UpdateExperimentStatus",
+            body: "*"
+        };
+    };
+
+
+    /**
+     * Register a Trial to DB.
+     * ID will be filled by manager automatically.
+     */
+    rpc RegisterTrial(RegisterTrialRequest) returns (RegisterTrialReply){
+        option (google.api.http) = {
+            post: "/api/Manager/RegisterTrial"
+            body: "trial"
+        };
+    };
+
+    /** 
+     * Get a list of Trial from DB by name of a Experiment.
+     */
+    rpc GetTrialList(GetTrialListRequest) returns (GetTrialListReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetTrialList/{experiment_name}"
+        };
+    };
+
+    /**
+     * Get a Trial from DB by ID of Trial.
+     */
+    rpc GetTrial(GetTrialRequest) returns (GetTrialReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetTrial/{trial_id}"
+        };
+    };
+
+    /** 
+     * Update Status of a trial.
+     */
+    rpc UpdateTrialStatus(UpdateTrialStatusRequest) returns (UpdateTrialStatusReply){
+        option (google.api.http) = {
+            put: "/api/Manager/UpdateTrialStatus",
+            body: "*"
+        };
+    };
+
+    /** 
+     * Report a log of Observations for a Trial.
+     * The log consists of timestamp and value of metric.
+     */
+    rpc ReportObservationLog(ReportObservationLogRequest) returns (ReportObservationLogReply){
+        option (google.api.http) = {
+            post: "/api/Manager/ReportObservationLog"
+            body: "*"
+        };
+    };
+
+    /**
+     * Get all log of Observations for a Trial.
+     */
+    rpc GetObservationLog(GetObservationLogRequest) returns (GetObservationLogReply){
+        option (google.api.http) = {
+            get: "/api/Manager/GetObservationLog/{trial_id}"
+            };
+    };
+
+    /** 
+     * Get Suggestions from a Suggestion service.
+     */
+    rpc GetSuggestions(GetSuggestionsRequest) returns (GetSuggestionsReply){
+        option (google.api.http) = {
+            post: "/api/Manager/GetSuggestions"
+            body: "*"
+        };
+    };
+
+    /** 
+     * Validate AlgorithmSettings in an Experiment.
+     * Suggestion service should return INVALID_ARGUMENT Error when the parameter is invalid
+     */
+     rpc ValidateAlgorithmSettings(ValidateAlgorithmSettingsRequest) returns (ValidateAlgorithmSettingsReply){
+        option (google.api.http) = {
+            post: "/api/Manager/ValidateAlgorithmSettings"
+            body: "*"
+        };
+    };
+}
+
+service Suggestion {
```

Comment:
Should we still name this service `Suggestion` or move to `Algorithm` ?

Reply:
I think we should remain the name `Suggestion Service`. 
The `Algorithm` means the algorithm for generating suggestions.
WDYT @richardsliu?

Reply:
Suggestion makes more sense. Algorithm is how suggestions are generated.